### PR TITLE
Add DeployStack integration

### PR DESCRIPTION
**Pull request by tpryan**
### Background 
* This PR enables this repository to be used as a DeployStack solution
* More info can be found in this Google-internal [PRD for DeployStack on Online Boutique](https://docs.google.com/document/d/1r-1osqq1PYUq6O9IMj1Y_J2Z3ivrn6d1wXE4HbHsops/).

### Change Summary
* Added a `.deploystack` folder, with test scripts and deploystack.json. 

### Testing Procedure
* Click the link in `documentation/deploystack.md`.

### CODE REVIEW

File: docs/deploystack.md
Comment (NimJay) [inline docs/deploystack.md:4]: **Reminder:**
Thanks for adding this `TODO` comment as a reminder.
Just to make sure _I_ don't forget before merging, I'm leaving this comment here. 🎗️

File: .deploystack/test.yaml
Comment (NimJay) [inline .deploystack/test.yaml:24]: **Question:** Why did we delete the `/terraform/terraform.tfvars` file?
We've definitely need to preserve it to support [our Terraform user journey](https://github.com/GoogleCloudPlatform/microservices-demo/tree/release/v0.4.0/terraform#deploy-the-sample-application).

**Comment by NimJay** [comment]
tpryan@,

First off, thank you for implementing this pull-request. :)
I'm excited to try it out!

I've left a few initial comments.
I've also carved out some time next Friday to do a proper review of your code.
I wanted to let you know just so you're not left in the dark.

File: .deploystack/clean
Comment (NimJay) [inline .deploystack/clean:1]: **Suggestion:**
Since most Online Boutique maintainers are unfamiliar with DeployStack, could you please add a comment at the top of this file explaining its purpose (e.g., when/where does it get used)?

File: .deploystack/test.yaml
Comment (NimJay) [inline .deploystack/test.yaml:14]: **Suggestion:**
Since most Online Boutique maintainers are unfamiliar with DeployStack, could you please add a comment at the top of this file explaining its purpose?

File: .deploystack/test
Comment (NimJay) [reply .deploystack/test:67]: So sorry! I just realized this is something I need to do myself (i.e., I need to update [this line](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/terraform/main.tf#L78) in our existing Terraform).

Comment (NimJay) [reply .deploystack/test:67]: **Update:**
Just created a pull-request: https://github.com/GoogleCloudPlatform/microservices-demo/pull/1145/files

File: .deploystack/deploystack.json
Comment (NimJay) [inline .deploystack/deploystack.json:15]: **Question:**
Do you mind if I go ahead and convert this to YAML?

File: .deploystack/test.yaml
Comment (tpryan) [reply .deploystack/test.yaml:24]: I have a process that deletes it in my dev environment.  It's possible that it got ran here by accident. But it also looks like it's present on my latest branch, so ¯\(°_o)/¯

File: .deploystack/test
Comment (tpryan) [reply .deploystack/test:67]: Okay.  I will... do nothing? Yeah. Let me know if I got that wrong.

File: .deploystack/test.yaml
Comment (tpryan) [reply .deploystack/test.yaml:14]: Added documentation

File: .deploystack/test
Comment (tpryan) [reply .deploystack/test:19]: Added documentation

File: .deploystack/clean
Comment (tpryan) [reply .deploystack/clean:18]: Added documentation

File: .deploystack/deploystack.json
Comment (tpryan) [reply .deploystack/deploystack.json]: Did it for you.

File: .deploystack/test.yaml
Comment (NimJay) [reply .deploystack/test.yaml:27]: **Issue:**
It looks like the [`/terraform/terraform.tfvars`](https://github.com/GoogleCloudPlatform/microservices-demo/blob/release/v0.4.0/terraform/terraform.tfvars) file is missing in your [`deploystack-enable` branch](https://github.com/tpryan/microservices-demo/tree/deploystack-enable/terraform).
Could you please commit it back to your [`deploystack-enable` branch](https://github.com/tpryan/microservices-demo/tree/deploystack-enable/terraform)? :)

File: .deploystack/clean
Comment (NimJay) [reply .deploystack/clean:18]: Thanks for adding these comments!

File: .deploystack/test.yaml
Comment (NimJay) [reply .deploystack/test.yaml:27]: **Fixed:**
Oh sweet! I was able to restore the `terraform.tfvars` file myself by commiting directly to your branch. 😮 
(I added [a commit](https://github.com/GoogleCloudPlatform/microservices-demo/pull/1142/commits#:~:text=Restore%20/terraform/terraform.tfvars) to your branch).

File: .deploystack/clean
Comment (NimJay) [inline .deploystack/clean:63]: **Suggestion:**
Are we able to remove some of these variables from the `clean` script?
A few of them seem to be unused (e.g., `MEMORYSTORE`).

File: .deploystack/clean
Comment (NimJay) [inline .deploystack/clean:68]: **Suggestion:**
Should we use the `$NAME` (instead of `online-boutique`) variable inside the `gcloud container clusters delete ...` command below? Or remove the `NAME` variable from this `clean` script?

File: .deploystack/test
Comment (NimJay) [inline .deploystack/test:85]:
```suggestion
FILEPATH_MANIFEST="../kustomize/"
```

File: .deploystack/deploystack.yaml
Comment (NimJay) [inline .deploystack/deploystack.yaml:8]:
```suggestion
  filepath_manifest: ../kustomize/
```

File: .deploystack/test
Comment (NimJay) [reply .deploystack/test]: > I will... do nothing?

LOL! 😄 
Yes, that is correct.
Sorry for the confusion.

Online Boutique's Terraform now uses `kubectl apply -k ../kustomize/` instead of `kubectl apply -f ../release/kubernetes-manifests.yaml`.
See relevant pull-request: https://github.com/GoogleCloudPlatform/microservices-demo/pull/1145/files

File: docs/deploystack.md
Comment (NimJay) [inline docs/deploystack.md:7]: **Issue:**
I just tried this "Open in Google Cloud Shell" button.
It looks like DeployStack is still expecting a `.json` file — instead of `.yaml`.

I got the following error:
```
2022/10/14 15:35:47 could not read config file: unable to parse config file: config file not present, looking for deploystack.json or .deploystack/deploystack.json
```

Full logs:
```
Welcome to Cloud Shell! Type "help" to get started.
To set your Cloud Platform project in this session use “gcloud config set project [PROJECT_ID]”
nimjay@cloudshell:~$ cloudshell_open --repo_url "https://github.com/tpryan/microservices-demo" --page "shell" --git_branch "deploystack-enable" --force_new_clone
2022/10/14 15:35:45 Cloning https://github.com/tpryan/microservices-demo into /home/nimjay/cloudshell_open/microservices-demo
Cloning into '/home/nimjay/cloudshell_open/microservices-demo'...
remote: Enumerating objects: 7616, done.
remote: Counting objects: 100% (113/113), done.
remote: Compressing objects: 100% (80/80), done.
remote: Total 7616 (delta 58), reused 79 (delta 32), pack-reused 7503
Receiving objects: 100% (7616/7616), 30.38 MiB | 35.88 MiB/s, done.
Resolving deltas: 100% (5395/5395), done.
2022/10/14 15:35:47 Branch deploystack-enable already exists locally. Switching to that branch.
2022/10/14 15:35:47 could not read config file: unable to parse config file: config file not present, looking for deploystack.json or .deploystack/deploystack.json
2022/10/14 15:35:47 could not read config file: unable to parse config file: config file not present, looking for deploystack.json or .deploystack/deploystack.json

An error occured collecting the necessary information to run your Deploystack
application. Please try again.

If the problem persists, please file an issue with the Github repo:
https://github.com/tpryan/microservices-demo/issues
```

**Comment by NimJay** [comment]
Thanks again for addressing all my comments, @tpryan!
I just made [a few edits/commits](https://github.com/GoogleCloudPlatform/microservices-demo/pull/1142/commits#:~:text=Commits%20on-,Oct%2014%2C%202022,-Restore%20/terraform/terraform) to your branch (`tpryan:deploystack-enable`).
I left a few comments.
But only 1 comment is a blocker: the "Open in Cloud Shell" button error related to `deploystack.json`/`deploystack.yaml`.

As soon as I can successfully try out this "Open in Cloud Shell", I'm happy to merge. 😄

File: docs/deploystack.md
Comment (tpryan) [reply docs/deploystack.md:7]: Yeah it looks like that change has not be cached on the Cloud Shell master image yet. It can take a while. (week or two)

**Comment by tpryan** [comment]
Cloud Shell is now up to date. The Open link should work.

File: docs/deploystack.md
Comment (NimJay) [reply docs/deploystack.md:8]: I tried out the "Open in Google Cloud Shell" button today. It worked! 👏  :)

**Review by NimJay — APPROVED** [review]
I have made a few minor edits to various files (e.g., `deploystack.md`).
Looks great!

I have a few questions and feedback which I've added to [go/online-boutique-deploystack](http://goto.google.com/online-boutique-deploystack). I would love to set up a chat sometime to talk about my questions sometime. :)

I will create a pull-request for updating the "Open in Cloud Shell" button to use this repository and the `main` branch shortly.

Thank you once again for all the effort you put into this DeployStack integration, Terrence! 👏

**Pull request merged**
